docs/source/advanced_topics/9_open_telemetry.rst

```diff
@@ -0,0 +1,78 @@
+
+More to say, the Trace is not a flat structure: a developer can wrap certain pieces of code with so called Spans, limiting the scope, so logs are attached to the hierarchy of spans rather than Trace directly. Every span is automatically a profiling object because OpenTelemetry collects its start time, end time and duration.
+
+Developers can attach to span auxiliary information: attributes, events, span statuses.
+
+Savant automatically creates spans for every pipeline stage and gives developer API to create nested spans in their Python code.
+
+OpenTelemetry send tracing information to a trace collector. Currently, Savant integrates with `Jaeger <https://www.jaegertracing.io/>`_: a widely-used open-source tracing solution. Jaeger not only gathers traces but has RESTful API and WebUI to access and analyze tracing information.
+
+.. image:: ../../../samples/telemetry/assets/01-trace.png
+
+OpenTelemetry Sampling
+^^^^^^^^^^^^^^^^^^^^^^
+
+Sampling is an approach of statistically limiting the number of messages based on 1 of every N principle.
+
+Depending on the sampling rate configured, the sampling fits both development/debug and production use. It allows combining code instrumenting with resource saving.
+
+A particular case is when the sampling rate is set to 0: the pipeline module does not create traces at all but still serves externally propagated traces.
+
+Trace Propagation
+^^^^^^^^^^^^^^^^^
+
+Trace propagation is a mechanism of passing traces between distributed, decoupled systems. Savant supports trace propagation.
+
+OpenTelemetry Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Use ``params.telemetry`` to configure OpenTelemetry for the module.
+
+.. code-block:: yaml
+
+    # base module parameters
+    parameters:
+      # DevMode (hot Python code reload on file change)
+      dev_mode: True
+
+      # enable OpenTelemetry
+      telemetry:
+        sampling_period: 100
+        root_span_name: pipeline
+        provider: jaeger
+        provider_params:
+          service_name: demo-pipeline
+          endpoint: jaeger:6831
+
+.. note::
+
+    The module `template <https://github.com/insight-platform/Savant/tree/develop/samples/template>`_ already has valid configuration, considering that the Jaeger is launched in the all-in-one mode recommended on the Jaeger `website <https://www.jaegertracing.io/docs/1.48/getting-started/>`_:
+
+    .. code-block:: bash
+
+        docker run -d --name jaeger \
+          -e COLLECTOR_ZIPKIN_HOST_PORT=:9411 \
+          -p 6831:6831/udp \
+          -p 6832:6832/udp \
+          -p 5778:5778 \
+          -p 16686:16686 \
+          -p 4317:4317 \
+          -p 4318:4318 \
+          -p 14250:14250 \
+          -p 14268:14268 \
+          -p 14269:14269 \
+          -p 9411:9411 \
+          jaegertracing/all-in-one:1.48
+
+**TODO**: do we need to link the dev container with Jaeger container somehow, or it is available?
```

Comment:
In general yes, the Jeager service needs to be accessible.

I've added 2 possible ways to connect module and client containers with the jaeger container to the template sample. Both are based on the named network configured in the docker compose file.

1. Run everything through docker compose.
2. Run only jaeger through docker compose, then run a dev container attached to the same named network.

Both ways allow to connect to jaeger using service name as host name.

Additionally, it's possible to start jaeger container as described in the comment above, but then add the  `--network host` flag to the module/client containers. Then the jaeger will be accessible on localhost